control-plane/pkg/kafka/clientpool/clientpool_test.go

```diff
@@ -117,6 +118,47 @@ func TestGetClusterAdmin(t *testing.T) {
 	cancel()
 }
 
+func TestClientCloses(t *testing.T) {
+	t.Parallel()
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+
+	cache := prober.NewLocalExpiringCache[clientKey, *client, struct{}](ctx, time.Second*1)
+	isClosed := atomic.NewBool(false)
+
+	clients := &ClientPool{
+		cache: cache,
+		newSaramaClient: func(_ []string, _ *sarama.Config) (sarama.Client, error) {
+			return &kafkatesting.MockKafkaClient{OnClose: func() {
+				isClosed.Toggle()
+			}}, nil
+		},
+		newClusterAdminFromClient: func(_ sarama.Client) (sarama.ClusterAdmin, error) {
+			return &kafkatesting.MockKafkaClusterAdmin{ExpectedTopics: []string{"topic1"}}, nil
+		},
+	}
+
+	client1, err := clients.GetClient(ctx, []string{"localhost:9092"}, nil)
+	assert.NoError(t, err)
+
+	clusterAdmin, err := clients.GetClusterAdmin(ctx, []string{"localhost:9092"}, nil)
+	assert.NoError(t, err)
+
+	clusterAdmin.Close()
+	client1.Close()
```

Comment:
Should the admin be also closed

With this, it's failing

```diff
diff --git a/control-plane/pkg/kafka/clientpool/clientpool_test.go b/control-plane/pkg/kafka/clientpool/clientpool_test.go
index 86c59908a..2ff1066ec 100644
--- a/control-plane/pkg/kafka/clientpool/clientpool_test.go
+++ b/control-plane/pkg/kafka/clientpool/clientpool_test.go
@@ -26,6 +26,7 @@ import (
        "go.uber.org/atomic"
        corev1 "k8s.io/api/core/v1"
        metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
        kafkatesting "knative.dev/eventing-kafka-broker/control-plane/pkg/kafka/testing"
        "knative.dev/eventing-kafka-broker/control-plane/pkg/prober"
 )
@@ -124,17 +125,23 @@ func TestClientCloses(t *testing.T) {
        ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
 
        cache := prober.NewLocalExpiringCache[clientKey, *client, struct{}](ctx, time.Second*1)
-       isClosed := atomic.NewBool(false)
+       clientClosed := atomic.NewBool(false)
+       adminClosed := atomic.NewBool(false)
 
        clients := &ClientPool{
                cache: cache,
                newSaramaClient: func(_ []string, _ *sarama.Config) (sarama.Client, error) {
                        return &kafkatesting.MockKafkaClient{OnClose: func() {
-                               isClosed.Toggle()
+                               clientClosed.Toggle()
                        }}, nil
                },
                newClusterAdminFromClient: func(_ sarama.Client) (sarama.ClusterAdmin, error) {
-                       return &kafkatesting.MockKafkaClusterAdmin{ExpectedTopics: []string{"topic1"}}, nil
+                       return &kafkatesting.MockKafkaClusterAdmin{
+                               ExpectedTopics: []string{"topic1"},
+                               OnClose: func() {
+                                       adminClosed.Toggle()
+                               },
+                       }, nil
                },
        }
 
@@ -149,11 +156,9 @@ func TestClientCloses(t *testing.T) {
 
        time.Sleep(time.Second * 2)
 
-       client2, err := clients.GetClient(ctx, []string{"localhost:9092"}, nil)
-       assert.NoError(t, err)
-
        // the client should have been closed successfully now
-       assert.True(t, isClosed.Load())
+       assert.True(t, clientClosed.Load())
+       assert.True(t, adminClosed.Load())
 
        client2.Close()
        cancel()
diff --git a/control-plane/pkg/kafka/testing/admin_mock.go b/control-plane/pkg/kafka/testing/admin_mock.go
index ec1c67287..0ab5cb5fd 100644
--- a/control-plane/pkg/kafka/testing/admin_mock.go
+++ b/control-plane/pkg/kafka/testing/admin_mock.go
@@ -44,6 +44,7 @@ type MockKafkaClusterAdmin struct {
        ErrorOnDeleteTopic error
 
        ExpectedClose      bool
+       OnClose            func()
        ExpectedCloseError error
 
        // DescribeTopics
@@ -301,5 +302,8 @@ func (m *MockKafkaClusterAdmin) RemoveMemberFromConsumerGroup(groupId string, gr
 
 func (m *MockKafkaClusterAdmin) Close() error {
        m.ExpectedClose = true
+       if m.OnClose != nil {
+               m.OnClose()
+       }
        return m.ExpectedCloseError
 }
```

Reply:
Oh that looks like a problem in our cluster admin proxy, let me fix that